File: codeBeispiel1/Applicative.java
import java.util.Optional;

public class Applicative{
    public static void main(String[] args) {
        Optional<Integer> maybeA = Optional.of(3);
        Optional<Integer> maybeB = Optional.of(5);
        // Anwendung von map und flatMap
        Optional<Integer> result = maybeA.flatMap(a ->
            maybeB.map(b -> a + b));
        System.out.println(result);
    }
}

File: codeBeispiel1/Applicative.hs
import Control.Applicative
-- Beispiel mit Maybe
maybeA :: Maybe Int
maybeA = Just 3
maybeB :: Maybe Int
maybeB = Just 5
result :: Maybe Int
-- Anwendung von pure und Verknüpfung
result = pure (+) <*> maybeA <*> maybeB
main :: IO()
main = do
    print result
